src/Swashbuckle.AspNetCore.SwaggerUI/CompressedEmbeddedFileResponser.cs

```diff
@@ -0,0 +1,186 @@
+﻿#nullable enable
+
+using Microsoft.AspNetCore.Http;
+using System.Reflection;
+using Microsoft.AspNetCore.StaticFiles;
+using System.IO.Compression;
+using System.Security.Cryptography;
+using Microsoft.Net.Http.Headers;
+
+#if NET
+using System.Collections.Frozen;
+#endif
+
+namespace Swashbuckle.AspNetCore;
+
+internal class CompressedEmbeddedFileResponser
+{
+    private readonly Assembly _assembly;
+    private readonly string _cacheControlHeaderValue;
+    private readonly string _pathPrefix;
+    private readonly FileExtensionContentTypeProvider _contentTypeProvider = new();
+
+#if NET
```

Comment:
There's a lot of these in this PR, so it's probably a good idea for me to get #3396 done first.

Reply:
FYI: #3422